src/payments-processors/lnbits-payment-processor.ts

```diff
@@ -0,0 +1,136 @@
+import { CreateInvoiceRequest, CreateInvoiceResponse, GetInvoiceResponse, IPaymentsProcessor } from '../@types/clients'
+import { deriveFromSecret, hmacSha256 } from '../utils/secret'
+import { Invoice, InvoiceStatus, InvoiceUnit } from '../@types/invoice'
+
+import { AxiosInstance } from 'axios'
+import { createLogger } from '../factories/logger-factory'
+import { Factory } from '../@types/base'
+import { Pubkey } from '../@types/base'
+import { Settings } from '../@types/settings'
+
+const debug = createLogger('lnbits-payments-processor')
+
+export class LNbitsInvoice implements Invoice {
+  id: string
+  pubkey: Pubkey
+  bolt11: string
+  amountRequested: bigint
+  amountPaid?: bigint
+  unit: InvoiceUnit
+  status: InvoiceStatus
+  description: string
+  confirmedAt?: Date | null
+  expiresAt: Date | null
+  updatedAt: Date
+  createdAt: Date
+}
+
+export class LNbitsCreateInvoiceResponse implements CreateInvoiceResponse {
+  id: string
+  pubkey: string
+  bolt11: string
+  amountRequested: bigint
+  description: string
+  unit: InvoiceUnit
+  status: InvoiceStatus
+  expiresAt: Date | null
+  confirmedAt?: Date | null
+  createdAt: Date
+  rawResponse?: string
+}
+
+export class LNbitsPaymentsProcesor implements IPaymentsProcessor {
+  public constructor(
+    private httpClient: AxiosInstance,
+    private settings: Factory<Settings>
+  ) {}
+
+  public async getInvoice(invoiceId: string): Promise<GetInvoiceResponse> {
+    debug('get invoice: %s', invoiceId)
+    try {
+      const response = await this.httpClient.get(`/api/v1/payments/${invoiceId}`, {
```

Comment:
May consider using `!` syntax to avoid errors if a value is `undefined` when using attribute accessors, like this:
`data.details!.paymentHash`. This is mostly in case LNBits schema changes, but we'll likely have bigger issues at that point.

https://www.typescriptlang.org/docs/handbook/release-notes/typescript-2-0.html#non-null-assertion-operator

Reply:
An alternative is to use json schema validation before doing the accessing of properties.